HCore-Database/ElasticSearch/Serializers/NewtonsoftSourceSerializer.cs

```diff
@@ -0,0 +1,151 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Elastic.Transport;
+using HCore.Database.ElasticSearch.JsonConverters;
+using HCore.Database.Extensions;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+
+namespace HCore.Database.ElasticSearch.Serializers
+{
+    public class NewtonsoftSourceSerializer : Serializer
+    {
+        internal const int _defaultBufferSize = 1024;
+
+        private readonly JsonSerializer _serializer;
+        private readonly JsonSerializer _collapsedSerializer;
+
+        private readonly JsonSerializerSettings _jsonSerializerSettings;
+
+        private readonly List<JsonConverter> _defaultConverters;
+        private readonly IEnumerable<JsonConverter> _jsonConverters;
+
+        public NewtonsoftSourceSerializer(Serializer builtInSerializer, JsonSerializerSettings jsonSerializerSettings = null, IEnumerable<JsonConverter> jsonConverters = null)
+        {
+            _jsonSerializerSettings = jsonSerializerSettings ?? new JsonSerializerSettings
+            {
+                ContractResolver = new CamelCasePropertyNamesContractResolver(),
```

Comment:
I had to add CamelCasePropertyNamesContractResolver, because their original implementation is way more complex and at some point they have this [NameMutator ](See here https://github.com/elastic/elasticsearch-net/blob/21e3758e90a90b1a5081c42ea76b4eca3c728307/src/Elasticsearch.Net/Utf8Json/Resolvers/DynamicObjectResolver.cs#L141) which produces similar end result